README.md

```diff
@@ -51,9 +52,74 @@ describe('MyWidget', () => {
 
 ## Vitest
 
-This package has an optional peer dependency on `vitest`. Any vitest-specific
-utility is exposed in a separate `@hypothesis/frontend-testing/vitest` entry
-point.
+All projects depending on this package are expected to use Vitest to run tests.
+
+However, we don't follow all practices recommended by Vitest. You should take
+these points into consideration when reading Vitest's documentation:
+
+1. By default, Vitest uses [Vite](https://vite.dev/) to bundle the tests and
+    source code as test files are run.
+    Instead, we manually pre-bundle our tests and source code in a single file
+    with Rollup, and that's the only test file we run.
+    That file is still processed by Vite, but since it has nothing to transform,
+    the time it takes is negligible.
+2. Due to the point above, Vitest tries to capture and calculate code coverage while
+    tests and sources are bundled with Vite.
+    In our case, we use `babel-plugin-istanbul` to instrument the code coverage
+    while we do our own pre-bundling.
+    This requires two configuration options to be set in `babel-plugin-istanbul`,
+    so that `istanbul-lib-instrument` checks the file is already instrumented.
+    See the details below.
```

Comment:
Can you add a link to the GitHub comment where this approach was suggested. I see the comment is mentioned in the code, but I think it would be worth linking here as well.